core/src/core/block.rs

```diff
@@ -168,11 +168,47 @@ impl Hashed for HeaderEntry {
 	}
 }
 
+/// Some type safety around header versioning.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize)]
+pub struct HeaderVersion(pub u16);
+
+impl Default for HeaderVersion {
```

Comment:
I'm not sure it makes sense to implement Default for HeaderVersion. In 2 months, all new blocks will have to be version 2, and having a default at that point seems like a good way to introduce logical errors.

Reply:
Default post-HF should be version 2 at that point. At least that's the intent here.
We may want to make it more explicit though and not use "default".

Reply:
Let's assume for a second that it will be version 2 post hard fork. How do you manage that? Do we want default to look up the current height to know if we're >= block 250000? I would assume no. My vote is to remove default.

Reply:
Was slightly agreeing with @DavidBurkett but then realized versions are rarely built by themselves, they're built with their headers. And then we have a bunch of code (mostly tests) that just want a header without caring much.

Reply:
The way I was thinking about this was in terms of two ways we need to create headers - 

* new/current headers - we want the default to be version 2 post hark fork
  * for tests (as @ignopeverell points out) it is reasonable to default to version 2 here
* historical headers (say during fast sync) - we want to create non-default headers taking height into consideration
  * any tests touching pre and post hard fork behavior will want to use non-default versions

But to be honest I have not thought too deeply about this - 
  * we only have a single version today
  * its not unreasonable to have a default now, until we decide otherwise
  * its easy to rework this as necessary in the future